frontend/src/components/NewRunParametersV2.tsx

```diff
@@ -257,18 +245,16 @@ function NewRunParametersV2(props: NewRunParametersProps) {
                       props.handleParameterChange(parametersInRealType);
                     }
 
-                    const { validInput, errorMessage } = checkInput(
+                    errorMessages[k] = generateInputValidationErrMsg(
                       parametersInRealType[k],
                       props.specParameters[k].parameterType,
                     );
-                    validInputs[k] = validInput;
-                    errorMessages[k] = errorMessage;
                     setErrorMessages(errorMessages);
-                    setValidInputs(validInputs);
 
-                    Object.values(validInputs).map(v2 => {
-                      allInputsValid = allInputsValid && v2;
+                    Object.values(errorMessages).map(v2 => {
+                      allInputsValid = allInputsValid && v2 === '';
```

Comment:
Is the `allInputsValid` still true when `v2` is null or undefined? `v2 === ''` is false in this case.

Reply:
Yes, it is. For this part, `allInputsValid` will be declared only when `onChange` has been called. Once the onChange is called, all values in errorMessages (i.e. each `v2`) will be either `errorMessage` or `empty string for (valid input)`. Thus, null or undefined shouldn't be in this case.

Reply:
Previously, I specify the return value of `generateInputValidationErrMsg()` must in string type, so `''` is return for valid input. I think your suggestion makes more sense! I will change the return value to `null` for valid input.